google/yaml_validator.rb

```diff
@@ -42,17 +42,40 @@ def set_variable(value, property)
       instance_variable_set("@#{property}", value)
     end
 
-    private
+    # Does all validation checking for a particular variable.
+    # options:
+    # :default   - the default value for this variable if its nil
+    # :type      - the allowed types (single or array) that this value can be
+    # :item_type - the allowed types that all values in this array should be
+    #              (impllied that type == array)
+    # :allowed   - the allowed values that this non-array variable should be.
+    # :required  - is the variable required? (defaults: true)
```

Comment:
I'd strongly prefer this to default to `false`; specifying `required: true` is a positive statement, and specifying `required: false` is negative. `optional: true` would also work, but more fields are optional than required so it seems like the stronger default.

Reply:
Yeah, I can do that.